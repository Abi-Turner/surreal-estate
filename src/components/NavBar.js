import "../styles/navbar.css";
import React from "react";
import { Link } from "react-router-dom";
import FacebookLogin from "react-facebook-login";
import logo from "../images/logo.png";

const NavBar = ({ onLogin, onLogout, userID }) => {
  return (
    <div className="nav-bar">
      <div className="logo-container">
        <img src={logo} alt="logo-icon" className="logo" />
        <h3 className="surreal-estate">Surreal Estate</h3>
      </div>
      <ul className="navbar-links">
        <li className="navbar-links-item">
          <Link className="item" to="/properties">
            View Properties
          </Link>
        </li>
        <li className="navbar-links-item">
          <Link className="item" to="/add-property">
            Add a Property
          </Link>
        </li>
        <li className="navbar-links-item">
          <Link className="item" to="/saved-properties">
            Saved Properties
          </Link>
        </li>
      </ul>
      <div className="facebook-login">
        {userID ? (
          <button
            type="submit"
            className="button-fb-signout"
            onClick={onLogout}
          >
            Sign Out
          </button>
        ) : (
          <FacebookLogin
            appId="188012377415038"
            callback={onLogin}
            icon="fa-facebook"
            autoLoad
          />
        )}
      </div>
    </div>
  );
};

export default NavBar;
