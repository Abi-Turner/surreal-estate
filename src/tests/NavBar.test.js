import { render } from "@testing-library/react";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import NavBar from "../components/NavBar";

jest.mock("react-facebook-login", () => {
  const FacebookLogin = () => <div />;
  return FacebookLogin;
});

describe("NavBar", () => {
  const { asFragment } = render(
    <BrowserRouter basename="/">
      <NavBar />
    </BrowserRouter>
  );

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
