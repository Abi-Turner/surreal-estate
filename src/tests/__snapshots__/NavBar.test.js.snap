// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="nav-bar"
  >
    <div
      class="logo-container"
    >
      <img
        alt="logo-icon"
        class="logo"
        src="logo.png"
      />
      <h3
        class="surreal-estate"
      >
        Surreal Estate
      </h3>
    </div>
    <ul
      class="navbar-links"
    >
      <li
        class="navbar-links-item"
      >
        <a
          class="item"
          href="/properties"
        >
          View Properties
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          class="item"
          href="/add-property"
        >
          Add a Property
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          class="item"
          href="/saved-properties"
        >
          Saved Properties
        </a>
      </li>
    </ul>
    <div
      class="facebook-login"
    >
      <div />
    </div>
  </div>
</DocumentFragment>
`;
